plugin/firewall/policy/engine.go

```diff
@@ -0,0 +1,71 @@
+package policy
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/request"
+)
+
+const (
+	// TypeNone policy action is UNKNOWN: no decision, apply the next rule.
+	TypeNone = iota
+	// TypeRefuse policy action is REFUSE: do not resolve a query and return code REFUSED
+	TypeRefuse
+	// TypeAllow policy action is ALLOW: continue to resolve query
+	TypeAllow
+	// TypeBlock policy action is BLOCK: do not resolve a query and return code NXDOMAIN
+	TypeBlock
+	// TypeDrop policy action is DROP: do not resolve a query and simulate a lost query
+	TypeDrop
+
+	// TypeCount total number of actions allowed
+	TypeCount
+)
+
+const (
+	nameTypeNone   = "none"
+	nameTypeAllow  = "allow"
+	nameTypeRefuse = "refuse"
+	nameTypeBlock  = "block"
+	nameTypeDrop   = "drop"
+)
+
+// NameTypes keep a mapping of the byte constant to the corresponding name
+var NameTypes [TypeCount]string
```

Comment:
this whole init section can be replaces with 
~~~
var NameType = [TypeCount]string{....}
~~~
the `nameTypeX` constant don't need to be defined, as they are only used here. Also wonder why you use an array and not an map? Having some inherit ordering is in the policy is also kind of confusing.

Reply:
Right. not sure why I started with an array. Will change and remove the nameXX constants